app.py

```diff
@@ -121,7 +134,9 @@ def graph_refresh_solver() -> None:
     count = 0
     for package_name, package_version, solver_name in packages:
         for index_url in indexes:
-            if _GRAPH_DB.python_package_version_exists(package_name, package_version, index_url, solver_name):
+            if _GRAPH_DB.python_package_version_exists(
+                package_name, package_version, index_url, solver_name
```

Comment:
I think we can discard this check and rather use `indexes` when `index_url` is set to `None` from  the knowledge graph query.